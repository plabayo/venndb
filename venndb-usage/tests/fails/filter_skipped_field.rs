use venndb::VennDB;

#[derive(Debug, VennDB)]
struct Employee {
    id: u32,
    is_manager: bool,
    #[venndb(skip)]
    is_active: bool,
}

fn main() {
    let mut db = EmployeeDB::new();
    db.append(Employee {
        id: 1,
        is_manager: true,
        is_active: true,
    });

    let mut query = db.query();
    query.is_active(true);
    assert!(query.execute().is_some());
}
