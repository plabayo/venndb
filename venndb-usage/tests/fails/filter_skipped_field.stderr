error[E0599]: no method named `is_active` found for struct `EmployeeDBQuery` in the current scope
  --> tests/fails/filter_skipped_field.rs:20:11
   |
3  | #[derive(Debug, VennDB)]
   |                 ------ method `is_active` not found for this struct
...
20 |     query.is_active(true);
   |           ^^^^^^^^^ method not found in `EmployeeDBQuery<'_>`
